src/coreclr/src/jit/lsra.cpp

```diff
@@ -6284,20 +6457,47 @@ void LinearScan::updateAssignedInterval(RegRecord* reg, Interval* interval, Regi
 #ifdef TARGET_ARM
     // Update overlapping floating point register for TYP_DOUBLE.
     Interval* oldAssignedInterval = reg->assignedInterval;
+    regNumber doubleReg           = REG_NA;
     if (regType == TYP_DOUBLE)
     {
-        RegRecord* anotherHalfReg = findAnotherHalfRegRec(reg);
-
+        doubleReg = reg->regNum;
+        assert(genIsValidDoubleReg(doubleReg));
+        RegRecord* anotherHalfReg        = getSecondHalfRegRec(reg);
```

Comment:
Wondering why don't we do something like `else` below i.e. call `findAnotherHalfRegRec()` instead of `getSecondHalfRegRec()`? Do things vary if `assignedInterval's registerType == TYP_DOUBLE` ?

Reply:
The comment of this method always expect `reg` to be lower half: `"reg" should be a even-numbered float register, i.e. lower half of double register.`